checkov/sast/engines/prisma_engine.py

```diff
@@ -0,0 +1,187 @@
+import ctypes
+import json
+import logging
+import os
+import platform
+import re
+import stat
+from pathlib import Path
+from typing import Optional, List, Set
+
+from checkov.common.bridgecrew.check_type import CheckType
+from checkov.common.bridgecrew.platform_integration import bc_integration
+from checkov.common.bridgecrew.platform_key import bridgecrew_dir
+from checkov.common.bridgecrew.severities import get_severity
+from checkov.common.models.enums import CheckResult
+from checkov.common.output.report import Report
+from checkov.common.typing import _CheckResult
+from checkov.common.util.http_utils import request_wrapper
+from checkov.sast.checks_infra.base_registry import Registry
+from checkov.sast.common import get_code_block
+from checkov.sast.consts import SastLanguages
+from checkov.sast.engines.base_engine import SastEngine
+from checkov.sast.prisma_models.report import PrismaReport
+from checkov.sast.record import SastRecord
+
+logger = logging.getLogger(__name__)
+
+
+class PrismaEngine(SastEngine):
+    def __init__(self) -> None:
+        self.lib_path = ""
+        self.check_type = CheckType.SAST
+        self.prisma_sast_dir_path = Path(bridgecrew_dir) / "sast"
+        self.sast_platform_base_path = "api/v1/sast"
+
+    def get_reports(self, targets: List[str], registry: Registry, languages: Set[SastLanguages]) -> List[Report]:
+        if not bc_integration.bc_api_key:
+            logging.info("The --bc-api-key flag needs to be set to run Sast prisma scanning")
+            return []
+
+        self.setup_sast_artifact()
+        prisma_lib_path = self.get_sast_artifact()
+        if not prisma_lib_path:
+            return []
+
+        self.lib_path = str(prisma_lib_path)
+
+        prisma_result = self.run_go_library(languages, source_codes=targets, policies=registry.checks_dirs_path)
+
+        return prisma_result
+
+    def setup_sast_artifact(self) -> bool:
+        current_version = ""
+        if not self.prisma_sast_dir_path.exists():
+            try:
+                os.makedirs(self.prisma_sast_dir_path, exist_ok=True)
+            except FileExistsError:
+                pass
+        else:
+            is_file_exists = [f for f in os.listdir(self.prisma_sast_dir_path) if
+                              (self.prisma_sast_dir_path / f).is_file() and "library" in f]
+            if len(is_file_exists) > 0:
+                match = re.match(r"(\d+_\d+_\d+)_library\.(so|dll)", is_file_exists[0])
+                if match:
+                    current_version = match.groups()[0]
+
+        status: bool = self.download_sast_artifacts(current_version)
+        return status
+
+    def download_sast_artifacts(self, current_version: str) -> bool:
```

Comment:
How do you feel about adding some kind of caching here like `@cached(TTLCache(maxsize=1, ttl=540))`?

Reply:
change it to 5m